streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java

```diff
@@ -1173,17 +1126,13 @@ public void shouldResetTransactionInitializedOnResetProducer() {
         );
         streamsProducer.initTransaction();
 
-        reset(mockedProducer);
-        mockedProducer.close();
-        mockedProducer.initTransactions();
```

Comment:
Correct me if I am wrong but easymock expected `mockedProducer.initTransactions();` to occur only one time reflected by expectLastCall() but it seems like it is being called twice for us when we moved to powermock. I am curious how was it working before?

Reply:
I believe the answer is that `streamsProducer.initTransaction` was called on line 1174. In EasyMock world we reset the mock and the counter went back to zero. In Mockito world we just keep incrementing the counter hence it gets bumped to 2. I am happy with either resetting the mock in Mockito world as well or writing a comment.

Reply:
I think the goal of this test is to make sure that `resetProducer` method resets `transactionInitialized` field so that subsequent call of the `initTransactions` method can start new transaction. Easymock expects  `mockedProducer.initTransactions()` only once because of the `reset()` on line 1176. As far as I know there is no equivalent of `reset()` in Mockito, so I think the best we can do is to check that `mockedProducer.initTransactions()` is called twice.

Reply:
Okay, but can you at least add this explanation in a condensed form above the verify(mockedProducer, ...).initTransactions() line?

Reply:
I've added a comment.